### PR TITLE
Gracefully close shovel if destination exchange not found

### PR DESCRIPTION
**Pull request by markus812498**
## Proposed Changes

This PR fixes a bug where if destination exchange aren't present, and prefetch_count is met with unacked messages, shovel won't automatically reconnect when destination exchange is created. I couldn't find a current issue # for it, but we came across it internally during a DR exercise.

Steps to reproduce (also included in test):
1. Create shovel, set prefetch=x, destination exchange to an non-existing exchange.
2. publish x messages
3. create the destination exchange. Shovel will need to be restarted for messages to move.

Originally we wanted to handle `?NOT_FOUND` similar to `?PRECONDITION_FAILED` in https://github.com/rabbitmq/rabbitmq-server/pull/5243, however using anonymous variable we can catch both in one line. Wdyt? 

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc.

### CODE REVIEW

**Comment by michaelklishin** [comment]
I cannot reproduce using `main` and the provided steps. Shovels always recover without manual intervention for me.

File: deps/rabbitmq_shovel/test/dynamic_SUITE.erl [hunk 1]
```diff
@@ -262,6 +263,36 @@ exchange(Config) ->
                              <<"queue">>, <<"hello">>)
       end).
 
+
+missing_dest_exchange(Config) ->
+    with_ch(Config,
+        fun (Ch) ->
+            amqp_channel:call(
+            Ch, #'queue.declare'{queue = <<"src">>, % E: record 'queue.declare' undefined
```
Comment (michaelklishin) [inline deps/rabbitmq_shovel/test/dynamic_SUITE.erl:271]: I'm not sure what these comments are meant to communicate?

**Review by michaelklishin — CHANGES_REQUESTED** [review]
Handling more error reasons seems OK to me but the comments
in the test confuse me and the code formatting does not match the rest of the module, or any well-known Erlang formatting convention I'm aware of.

**Comment by michaelklishin** [comment]
I addressed the formatting and misleading comments part in #8697.

File: deps/rabbitmq_shovel/test/dynamic_SUITE.erl (hunk 1 above)
Comment (markus812498) [reply deps/rabbitmq_shovel/test/dynamic_SUITE.erl:271]: Thank you and sorry about that, it looks like remains of a poorly configured linter on my end. Should've spotted that!

**Pull request merged**